app/assets/javascripts/discourse/routes/build-topic-route.js.es6

```diff
@@ -91,7 +91,7 @@ export default function(filter, extras) {
       const topicOpts = {
         model,
         category: null,
-        period: model.get('for_period') || (filter.indexOf('/') > 0 ? filter.split('/')[1] : ''),
+        period: model.get('for_period') || (filter.indexOf('top/') >= 0 ? filter.split('/')[1] : ''),
```

Comment:
hmmm what is this change for?

Reply:
I had to add a few new topic lists: favorites/latest, favorites/new, favorites/unread. When I added these, the filter would be misparsed and consider 'latest', 'new' or 'unread' as period which was wrong. The timeperiod should only be extracted when the filter is a 'top' filter (it contains 'top/').

Reply:
@nbianca It'll be better to move this fix into another PR or at least into another commit. Otherwise, the commit is actually doing more than what the commit message say it does.

Reply:
@tgxworld Done, see #5614.